src/store.ts

```diff
@@ -49,14 +66,11 @@ class CookieStore {
    * Respects the `request.credentials` policy.
    */
   add(request: RequestLike, response: ResponseLike): void {
-    try {
-      if (request.credentials === 'omit') {
-        return
-      }
-    } catch {
-      // Ignore errors thrown by `request.credentials` access, such as those in cloudflare workers
-      // We can't just check that `credentials` is in request, because it is in cloudflare workers test environment
-      // however access throws an error
+    if (
```

Comment:
I'd split this `if` into two as they handle 2 distinct use cases. What do you think?

Reply:
Why not:

```js
// Ignore the cases when "request.credentials" is not implemented.
if (!isPropertyAccessible(request, 'credentials')) {
  return 
}

// Handle the scenario when "credentials" was set to "omit".
if (request.credentials === 'omit') {
  return
}
```

Reply:
@kettanaito that check is diifferent than the one implemented here. 

We _do_ want to allow interacting with cookies when credentials are not implemented. We _only want to check for omit_ when they _are_ implemented. 

the logic here is

- if credentials can be accessed _and_ they are omit, don't interact
- if credentials is NOT implemented, or credentials are omit, return early

In your change we would return early if credentials are not implemented OR if they are omit. Which would fail to modify cookies in the case credentials are not implemented, which is not what we want in this case (as far as I can tell from a quick skimming of the docs on cloudflare - it is fine to interact with cookies there, there's just not credential check)

Reply:
Oh, I see now. Apologies, took me a minute to wrap my head around this. Please, feel free to proceed with your latest suggestion, it looks great then.